worker/src/backgroundMigrations/backgroundMigrationManager.ts

```diff
@@ -0,0 +1,178 @@
+          },
+        );
+
+        if (!BackgroundMigrationManager.activeMigration) {
+          continue;
+        }
+
+        // Initiate heartbeats every couple seconds
+        await BackgroundMigrationManager.heartBeat();
+
+        const { migration, args } = BackgroundMigrationManager.activeMigration;
+        const { valid, invalidReason } = await migration.validate(args);
+        if (!valid) {
+          logger.error(
+            `Validation failed for background migration ${BackgroundMigrationManager.activeMigration.name}: ${invalidReason}`,
+          );
+          await prisma.backgroundMigration.update({
+            where: {
+              id: BackgroundMigrationManager.activeMigration.id,
+              workerId: BackgroundMigrationManager.workerId,
+            },
+            data: {
+              lockedAt: null,
+              failedAt: new Date(),
+              failedReason: invalidReason,
+            },
+          });
+          continue;
+        }
+
+        try {
+          await migration.run(args);
+
+          if (BackgroundMigrationManager.activeMigration !== undefined) {
+            // Only mark as complete if still active. Otherwise, it was aborted.
+            await prisma.backgroundMigration.update({
+              where: {
+                id: BackgroundMigrationManager.activeMigration.id,
+                workerId: BackgroundMigrationManager.workerId,
+              },
+              data: {
+                finishedAt: new Date(),
+                lockedAt: null,
+              },
+            });
+            logger.info(
+              `Finished background migration ${BackgroundMigrationManager.activeMigration.name}`,
+            );
+          }
+        } catch (err) {
+          logger.error(
+            `Failed to run background migration ${BackgroundMigrationManager.activeMigration.name}: ${err}`,
+          );
+          await prisma.backgroundMigration.update({
+            where: {
+              id: BackgroundMigrationManager.activeMigration.id,
+              workerId: BackgroundMigrationManager.workerId,
+            },
+            data: {
+              lockedAt: null,
+              failedAt: new Date(),
+              failedReason:
+                err instanceof Error ? err.message : "Unknown error",
+            },
+          });
+        }
+        BackgroundMigrationManager.activeMigration = undefined;
+      }
+    });
+  }
+
+  public static async close(): Promise<void> {
+    if (BackgroundMigrationManager.activeMigration) {
+      await BackgroundMigrationManager.activeMigration.migration.abort();
+      await prisma.backgroundMigration.update({
+        where: {
+          id: BackgroundMigrationManager.activeMigration.id,
+          workerId: BackgroundMigrationManager.workerId,
+        },
+        data: {
+          lockedAt: null,
+        },
+      });
+      BackgroundMigrationManager.activeMigration = undefined;
+    }
```

Comment:
should we add a log line here that the active migration was aborted successfully?